### DISCUSSION TITLE
Keyboard not working when osu! Laser is in fullscreen on mac

### DISCUSSION THREAD

**Discussion by Xencein** [ppy/osu#15490] [Q&A] [answered]
The keyboard works when in windowed or borderless screen.
In fullscreen mode the the key works except when typing something.(Eg:- The keys work when playing the game or when using a shortcut such as Ctrl+Shift+R, but does not work when searching for a beatmap or in the options menu.
I play on a M1 mac base version.

https://user-images.githubusercontent.com/93763522/140489049-80cad15b-470a-4e63-b9d6-2bc8f1e48fda.mp4

**Comment by frenzibyte** [comment]
The fact that you can interact with the game just fine but not be able to type to textboxes means this may have to do with the `TextBox`'s `ITextInputSource`-related code. I cannot reproduce this, however.

**Reply by Xencein** [reply]
Yeah, sorry my bad, my language was set to Romanji and I did not realize it, its fixed now.
Thanks for Helping

**Reply by peppy** [reply]
It's 99% related to the user have a custom input IME setup. @Xencein can you confirm it works when you set your system input to English? Also can you share which input method you're using when it doesn't work?

**Comment by peppy** [comment, marked as answer]
Tracking via https://github.com/ppy/osu/issues/14747.
